From 1786e94c332ca67da98ec95e4a67df86962cb91c Mon Sep 17 00:00:00 2001
From: yuanxia <yuanxia2073@phytium.com.cn>
Date: Tue, 21 May 2024 16:39:58 +0800
Subject: [PATCH 061/360] dt-bindings: can: Add bindings for Phytium CAN

This patch documents the DT bindings for Phytium CAN controller.

Signed-off-by: yuanxia <yuanxia2073@phytium.com.cn>
Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
---
 .../bindings/net/can/phytium,can.yaml         | 72 +++++++++++++++++++
 1 file changed, 72 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/can/phytium,can.yaml

diff --git a/Documentation/devicetree/bindings/net/can/phytium,can.yaml b/Documentation/devicetree/bindings/net/can/phytium,can.yaml
new file mode 100644
index 000000000000..dd4b8000830c
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/can/phytium,can.yaml
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/net/can/phytium,can.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title:
+  Phytium CAN controller
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+properties:
+  compatible:
+    enum:
+      - phytium,can
+      - phytium,canfd
+
+  reg:
+    maxItems: 1
+    description: Should contain the controller registers location and length
+
+  interrupts:
+    maxItems: 1
+    description: Should contain IRQ line for the controller
+
+  clocks:
+    maxItems: 1
+    description: Input clock used by the controller
+
+  clock-names:
+    maxItems: 1
+    description: Input clock name, should be "can_clk"
+    const: can_clk
+
+  tx-fifo-depth:
+    $ref: "/schemas/types.yaml#/definitions/uint32"
+    description: Indicates the length of TX FIFO
+
+  rx-fifo-depth:
+    $ref: "/schemas/types.yaml#/definitions/uint32"
+    description: Indicates the length of RX FIFO
+
+  extend_brp:
+    description: |
+      Indicates to apply the extend BRP parameter of bit timming for
+      early version of CAN controller
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - clocks
+  - clock-names
+  - tx-fifo-depth
+  - rx-fifo-depth
+
+allOf:
+  - $ref: can-controller.yaml#
+
+examples:
+  - |
+
+    can0: can@2800a000 {
+      compatible = "phytium,canfd";
+      reg = <0x0 0x2800a000 0x0 0x1000>;
+      interrupts = <GIC_SPI 81 IRQ_TYPE_LEVEL_HIGH>;
+      clocks = <&sysclk_600mhz>;
+      clock-names = "can_clk";
+      tx-fifo-depth = <64>;
+      rx-fifo-depth = <64>;
+    };
-- 
2.20.1

